### PR TITLE
mao: add support for 'None' provider type

### PR DESCRIPTION
**Pull request by michaelgugino**
Fixes: https://bugzilla.redhat.com/show_bug.cgi?id=1691120

### CODE REVIEW

File: pkg/operator/config.go [hunk 1]
```diff
@@ -81,6 +82,9 @@ type InstallPlatform struct {
 
 	// BareMetal is the configuration used when running on managed Bare Metal
 	BareMetal interface{} `json:"baremetal,omitempty"`
+
+	// None is the configuration used when running on unmanaged UPI
+	None interface{} `json:"baremetal,omitempty"`
```
Comment (abhinavdahiya) [inline pkg/operator/config.go]: `json:"none,omitempty"`

File: pkg/operator/operator.go [hunk 2]
```diff
@@ -176,7 +176,14 @@ func (optr *Operator) sync(key string) error {
 		glog.Errorf("Failed getting operator config: %v", err)
 		return err
 	}
-
+	if operatorConfig.Controllers.Provider == "None" {
+		glog.V(1).Info("`None` provider specified, reporting available")
+		if err := optr.statusAvailable(); err != nil {
+			glog.Errorf("Error syncing ClusterOperatorStatus: %v", err)
+			return fmt.Errorf("error syncing ClusterOperatorStatus: %v", err)
```
Comment (enxebre) [inline pkg/operator/operator.go]: @michaelgugino should we `optr.statusFailing` here?

Comment (michaelgugino) [reply pkg/operator/operator.go]: I don't think so.  If we fail to set statusAvailable it's most likely due to temporary network outage / api unavailable, so we want to retry I think.

Generally, if provider is None type, I think we want to do a no-op and report available.

**Pull request merged**